File: Romania/Initializer.java
import java.util.*;

public class Initializer {

	public static Map<String, ArrayList<Neighbor>> MapInitializer(Map<String, ArrayList<Neighbor>> map) {
		map.put("Arad", new ArrayList<Neighbor>(Arrays.asList(new Neighbor("Zerind", 75), new Neighbor("Sibiu", 140), new Neighbor("Timisoara", 118))));
		map.put("Zerind", new ArrayList<Neighbor>(Arrays.asList(new Neighbor("Arad", 75), new Neighbor("Oradea", 71))));
		map.put("Timisoara", new ArrayList<Neighbor>(Arrays.asList(new Neighbor("Arad", 118), new Neighbor("Lugoj", 111))));
		map.put("Lugoj", new ArrayList<Neighbor>(Arrays.asList(new Neighbor("Timisoara", 111), new Neighbor("Mehadia", 70))));
		map.put("Oradea", new ArrayList<Neighbor>(Arrays.asList(new Neighbor("Zerind", 71), new Neighbor("Sibiu", 151))));
		map.put("Sibiu", new ArrayList<Neighbor>(Arrays.asList(new Neighbor("Arad", 140), new Neighbor("Oradea", 151), new Neighbor("Rimnicu Vilcea", 80), new Neighbor("Fagaras", 99))));
		map.put("Mehadia", new ArrayList<Neighbor>(Arrays.asList(new Neighbor("Lugoj", 70), new Neighbor("Drobeta", 75))));
		map.put("Drobeta", new ArrayList<Neighbor>(Arrays.asList(new Neighbor("Mehadia", 75), new Neighbor("Craiova", 120))));
		map.put("Craiova", new ArrayList<Neighbor>(Arrays.asList(new Neighbor("Drobeta", 120), new Neighbor("Rimnicu Vilcea", 146), new Neighbor("Pitesti", 138))));
		map.put("Rimnicu Vilcea", new ArrayList<Neighbor>(Arrays.asList(new Neighbor("Sibiu", 80), new Neighbor("Craiova", 146), new Neighbor("Pitesti", 97))));
		map.put("Fagaras", new ArrayList<Neighbor>(Arrays.asList(new Neighbor("Sibiu", 99), new Neighbor("Bucharest", 211))));
		map.put("Pitesti", new ArrayList<Neighbor>(Arrays.asList(new Neighbor("Rimnicu Vilcea", 97), new Neighbor("Craiova", 138), new Neighbor("Bucharest", 101))));
		map.put("Bucharest", new ArrayList<Neighbor>(Arrays.asList(new Neighbor("Pitesti", 101), new Neighbor("Fagaras", 211), new Neighbor("Giurgiu", 90), new Neighbor("Urziceni", 85))));
		map.put("Giurgiu", new ArrayList<Neighbor>(Arrays.asList(new Neighbor("Bucharest", 90))));
		map.put("Urziceni", new ArrayList<Neighbor>(Arrays.asList(new Neighbor("Bucharest", 85), new Neighbor("Hirsova", 98), new Neighbor("Vaslui", 142))));
		map.put("Hirsova", new ArrayList<Neighbor>(Arrays.asList(new Neighbor("Urziceni", 98), new Neighbor("Eforie", 86))));
		map.put("Eforie", new ArrayList<Neighbor>(Arrays.asList(new Neighbor("Hirsova", 86))));
		map.put("Vaslui", new ArrayList<Neighbor>(Arrays.asList(new Neighbor("Urziceni", 142), new Neighbor("Iasi", 92))));
		map.put("Iasi", new ArrayList<Neighbor>(Arrays.asList(new Neighbor("Vaslui", 92), new Neighbor("Neamt", 87))));
		map.put("Neamt", new ArrayList<Neighbor>(Arrays.asList(new Neighbor("Iasi", 87))));

		return map;
	}

	public static Map<String, Boolean> VisitedInitializer(Map<String, Boolean> visited) {
		visited.put("Arad", false);
		visited.put("Zerind", false);
		visited.put("Timisoara", false);
		visited.put("Lugoj", false);
		visited.put("Oradea", false);
		visited.put("Sibiu", false);
		visited.put("Mehadia", false);
		visited.put("Drobeta", false);
		visited.put("Craiova", false);
		visited.put("Rimnicu Vilcea", false);
		visited.put("Fagaras", false);
		visited.put("Pitesti", false);
		visited.put("Bucharest", false);
		visited.put("Giurgiu", false);
		visited.put("Urziceni", false);
		visited.put("Hirsova", false);
		visited.put("Eforie", false);
		visited.put("Vaslui", false);
		visited.put("Iasi", false);
		visited.put("Neamt", false);

		return visited;
	}
}

File: Romania/cs/Initializer.cs
using System;
using System.Collections.Generic;

namespace Romania
{
    class Initializer
    {
        public static Dictionary<String, State> MapInitializer(Dictionary<String, State> graph)
        {
            graph.Add("Arad", new State("Arad", new List<string>(new []{"Zerind", "Sibiu", "Timisoara"}), new List<int>(new []{75, 140, 118})));
            graph.Add("Zerind", new State("Zerind", new List<string>(new []{"Arad", "Oradea"}), new List<int>(new []{75, 71})));
            graph.Add("Timisoara", new State("Timisoara", new List<string>(new []{"Arad", "Lugoj"}), new List<int>(new []{118, 111})));
            graph.Add("Lugoj", new State("Lugoj", new List<string>(new []{"Timisoara", "Mehadia"}), new List<int>(new []{111, 70})));
            graph.Add("Oradea", new State("Oradea", new List<string>(new []{"Zerind", "Sibiu"}), new List<int>(new []{71, 151})));
            graph.Add("Sibiu", new State("Sibiu", new List<string>(new []{"Arad", "Oradea", "Rimnicu Vilcea", "Fagaras"}), new List<int>(new []{140, 151, 80, 99})));
            graph.Add("Mehadia", new State("Mehadia", new List<string>(new []{"Lugoj", "Drobeta"}), new List<int>(new []{70, 75})));
            graph.Add("Drobeta", new State("Drobeta", new List<string>(new []{"Mehadia", "Craiova"}), new List<int>(new []{75, 120})));
            graph.Add("Craiova", new State("Craiova", new List<string>(new []{"Drobeta", "Rimnicu Vilcea", "Pitesti"}), new List<int>(new []{120, 146, 138})));
            graph.Add("Rimnicu Vilcea", new State("Rimnicu Vilcea", new List<string>(new []{"Sibiu", "Craiova", "Pitesti"}), new List<int>(new []{80, 146, 97})));
            graph.Add("Fagaras", new State("Fagaras", new List<string>(new []{"Sibiu", "Bucharest"}), new List<int>(new []{99, 211})));
            graph.Add("Pitesti", new State("Pitesti", new List<string>(new []{"Rimnicu Vilcea", "Craiova", "Bucharest"}), new List<int>(new []{75, 71})));
            graph.Add("Bucharest", new State("Bucharest", new List<string>(new []{"Pitesti", "Fagaras", "Giurgiu", "Urziceni"}), new List<int>(new []{101, 211, 90, 85})));
            graph.Add("Giurgiu", new State("Giurgiu", new List<string>(new []{"Bucharest"}), new List<int>(new []{90})));
            graph.Add("Urziceni", new State("Urziceni", new List<string>(new []{"Bucharest", "Hirsova", "Vaslui"}), new List<int>(new []{85, 98, 142})));
            graph.Add("Hirsova", new State("Hirsova", new List<string>(new []{"Urziceni", "Eforie"}), new List<int>(new []{98, 86})));
            graph.Add("Eforie", new State("Eforie", new List<string>(new []{"Hirsova"}), new List<int>(new []{86})));
            graph.Add("Vaslui", new State("Vaslui", new List<string>(new []{"Urziceni"}), new List<int>(new []{142})));
            graph.Add("Iasi", new State("Iasi", new List<string>(new []{"Vaslui", "Neamt"}), new List<int>(new []{92, 87})));
            graph.Add("Neamt", new State("Neamt", new List<string>(new []{"Iasi"}), new List<int>(new []{87})));

            return graph;
        }
    }
}
